File: Problems/118-pascals-triangle/pascals-triangle.py
class Solution:
    def generate(self, numRows: int) -> List[List[int]]:
        ans = [[1]]
        for i in range(numRows-1):
            l = ans[-1].copy()
            ans.append([])
            for ind,v in enumerate(l,start =-1):
                if ind<0:
                    ans[-1].append(v)
                else:
                    ans[-1].append(v+l[ind])
            ans[-1].append(1)
        return ans
                    

File: Problems/118-pascals-triangle/pascals-triangle.cpp
class Solution {
public:
    vector<vector<int>> generate(int numRows) {
        vector<vector<int>>v(numRows);
        v[0].push_back(1);
        for(int i=1;i<numRows;i++)
        {
             v[i].push_back(1);
             for(int j=1;j<i;j++)v[i].push_back(v[i-1][j]+v[i-1][j-1]);
             v[i].push_back(1);
        }
        return v;
    }
};